File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/RemoveObsoleteCode.cs
using System;

namespace Tests
{
    [Obsolete] // Noncompliant {{Do not forget to remove this deprecated code someday.}}
//   ^^^^^^^^
    public class Program
    {
        [Obsolete("Message")]                // Noncompliant
        public delegate void CloseDelegate(object sender, EventArgs eventArgs);

        [Obsolete("Message", error: true)]   // Noncompliant
        public event CloseDelegate OnClose;

        [ObsoleteAttribute()]                // Noncompliant
        public Program()
        {

        }
    }
}

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/RemoveObsoleteCode.vb
Imports System

<Obsolete> ' Noncompliant ^2#8 {{Do not forget to remove this deprecated code someday.}}
Public Class Program

    <Obsolete("Message")>                   ' Noncompliant
    Public Delegate Sub CloseDelegate(sender As Object, eventArgs As EventArgs)

    <Obsolete("Message", True)>             ' Noncompliant
    Public Event OnClose As CloseDelegate

    <ObsoleteAttribute()>                   ' Noncompliant
    Public Sub New()
    End Sub
End Class
